core/src/test/java/org/elasticsearch/search/aggregations/metrics/geocentroid/InternalGeoCentroidTests.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.search.aggregations.metrics.geocentroid;
+
+import org.apache.lucene.spatial.geopoint.document.GeoPointField;
+import org.elasticsearch.common.geo.GeoPoint;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.search.aggregations.InternalAggregationTestCase;
+import org.elasticsearch.search.aggregations.pipeline.PipelineAggregator;
+import org.elasticsearch.test.geo.RandomGeoGenerator;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public class InternalGeoCentroidTests extends InternalAggregationTestCase<InternalGeoCentroid> {
+
+    @Override
+    protected InternalGeoCentroid createTestInstance(String name, List<PipelineAggregator> pipelineAggregators,
+                                                     Map<String, Object> metaData) {
+        GeoPoint centroid = RandomGeoGenerator.randomPoint(random());
+        long encodedGeoPoint = GeoPointField.encodeLatLon(centroid.lat(), centroid.lon());
+        centroid = new GeoPoint(GeoPointField.decodeLatitude(encodedGeoPoint), GeoPointField.decodeLongitude(encodedGeoPoint));
```

Comment:
Right, we can reuse the current one and just set lat/lon on that. Let me change that.